controllers/llamastackdistribution_controller.go

```diff
@@ -162,45 +215,303 @@ func (r *LlamaStackDistributionReconciler) reconcileResources(ctx context.Contex
+			builder.WithPredicates(predicate.Funcs{
+				UpdateFunc: r.configMapUpdatePredicate,
+				CreateFunc: r.configMapCreatePredicate,
+				DeleteFunc: r.configMapDeletePredicate,
+			}),
+		).
 		Complete(r)
 }
 
+// createConfigMapFieldIndexer creates a field indexer for ConfigMap references.
+func (r *LlamaStackDistributionReconciler) createConfigMapFieldIndexer(ctx context.Context, mgr ctrl.Manager) error {
+	if err := mgr.GetFieldIndexer().IndexField(
+		ctx,
+		&llamav1alpha1.LlamaStackDistribution{},
+		"spec.server.userConfig.configMapName",
+		r.configMapIndexFunc,
+	); err != nil {
+		return fmt.Errorf("failed to create ConfigMap reference field indexer: %w", err)
+	}
+	return nil
+}
+
+// configMapIndexFunc is the indexer function for ConfigMap references.
+func (r *LlamaStackDistributionReconciler) configMapIndexFunc(rawObj client.Object) []string {
+	llsd, ok := rawObj.(*llamav1alpha1.LlamaStackDistribution)
+	if !ok {
+		return nil
+	}
+	if !hasValidUserConfig(llsd) {
+		return nil
+	}
+
+	// Create index key as "namespace/name" format
+	configMapNamespace := getUserConfigMapNamespaceStandalone(llsd)
+	indexKey := fmt.Sprintf("%s/%s", configMapNamespace, llsd.Spec.Server.UserConfig.ConfigMapName)
+	return []string{indexKey}
+}
+
+// llamaStackUpdatePredicate returns a predicate function for LlamaStackDistribution updates.
+func (r *LlamaStackDistributionReconciler) llamaStackUpdatePredicate(mgr ctrl.Manager) func(event.UpdateEvent) bool {
+	return func(e event.UpdateEvent) bool {
+		// Safely type assert old object
+		oldObj, ok := e.ObjectOld.(*llamav1alpha1.LlamaStackDistribution)
+		if !ok {
+			return false
+		}
+		oldObjCopy := oldObj.DeepCopy()
+
+		// Safely type assert new object
+		newObj, ok := e.ObjectNew.(*llamav1alpha1.LlamaStackDistribution)
+		if !ok {
+			return false
+		}
+		newObjCopy := newObj.DeepCopy()
+
+		// Compare only spec, ignoring metadata and status
+		if diff := cmp.Diff(oldObjCopy.Spec, newObjCopy.Spec); diff != "" {
+			logger := mgr.GetLogger().WithValues("namespace", newObjCopy.Namespace, "name", newObjCopy.Name)
+			logger.Info("LlamaStackDistribution CR spec changed")
+			// Note that both the logger and fmt.Printf could appear entangled in the output
+			// but there is no simple way to avoid this (forcing the logger to flush its output).
+			// When the logger is used to print the diff the output is hard to read,
+			// fmt.Printf is better for readability.
+			fmt.Printf("%s\n", diff)
+		}
+
+		return true
+	}
+}
+
+// configMapUpdatePredicate handles ConfigMap update events.
+func (r *LlamaStackDistributionReconciler) configMapUpdatePredicate(e event.UpdateEvent) bool {
+	oldConfigMap, oldOk := e.ObjectOld.(*corev1.ConfigMap)
+	newConfigMap, newOk := e.ObjectNew.(*corev1.ConfigMap)
+
+	if !oldOk || !newOk {
+		return false
+	}
+
+	// Only trigger if Data or BinaryData has changed
+	dataChanged := !cmp.Equal(oldConfigMap.Data, newConfigMap.Data)
```

Comment:
Can we print the diff?

Reply:
Can do! I've added diff printing for ConfigMap Data when changes are detected.

Example:
```
2025-06-27T21:38:09Z	INFO	Referenced ConfigMap change detected	{"configMapName": "llama-stack-config", "configMapNamespace": "new-llama"}
2025-06-27T21:38:09Z	INFO	ConfigMap Data changed	{"configMapName": "llama-stack-config", "configMapNamespace": "new-llama"}
ConfigMap new-llama/llama-stack-config Data diff:
  map[string]string{
  	"run.yaml": (
  		"""
  		... // 3 identical lines
  		apis:
  		- inference
+ 		- scoring
  		providers:
  		  inference:
  		... // 11 identical lines
  		"""
  	),
  }
```